api/python/quilt3/util.py

```diff
@@ -140,7 +144,9 @@ def __init__(self, bucket, path, version_id):
             if os.name == 'nt':
                 assert '\\' not in path, "Paths must use / as a separator"
         else:
-            assert not path.startswith('/'), "S3 paths must not start with '/'"
+            if path.startswith('/'):
+                # S3 paths are not absolute, but there may be a "/" folder in root.
+                logger.warning("S3 paths should not start with '/': %r", path)
```

Comment:
do we really need this warning given it's seems to be a valid case?
I think we had that assert to make sure we parse S3 URIs in `from_url()` correctly, but now that seems too defensive